schematics/src/scope/index.ts

```diff
@@ -0,0 +1,81 @@
+import { dasherize } from '@angular-devkit/core/src/utils/strings';
+import {
+  Rule,
+  Tree,
+  SchematicContext,
+  externalSchematic,
+  Source,
+  mergeWith,
+  chain,
+  noop
+} from '@angular-devkit/schematics';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { ScriptTarget, createSourceFile } from 'typescript';
+import { applyChanges } from '../ng-add';
+import { LIB_NAME } from '../schematics.consts';
+import { addProviderToModule, insertImport, addImportToModule } from '../utils/ast-utils';
+import { findModuleFromOptions } from '../utils/find-module';
+import { getProject, getProjectPath } from '../utils/projects';
+import { createTranslateFilesFromOptions } from '../utils/translations';
+import { SchemaOptions } from './schema';
+
+const p = require('path');
+
+function addScopeToModule(tree: Tree, modulePath: string, name: string) {
+  const module = tree.read(modulePath);
+
+  const moduleSource = createSourceFile(modulePath, module.toString('utf-8'), ScriptTarget.Latest, true);
+  const provider = `{ provide: TRANSLOCO_SCOPE, useValue: '${dasherize(name)}' }`;
+  const changes = [];
+  changes.push(addProviderToModule(moduleSource, modulePath, provider, LIB_NAME)[0]);
+  changes.push(addImportToModule(moduleSource, modulePath, 'TranslocoModule', LIB_NAME)[0]);
+  changes.push(insertImport(moduleSource, modulePath, 'TRANSLOCO_SCOPE, TranslocoModule', LIB_NAME));
+
+  applyChanges(tree, modulePath, changes as any);
+}
+
+function getTranslationFilesFromAssets(host, sourceRoot) {
+  const langFiles = host.root.dir((sourceRoot + '/assets/i18n') as any).subfiles;

```

Comment:
You assume that everyone will use `/assets/i18n`, but it may not be the case. You should allow passing a different path.